monai/transforms/transforms.py

```diff
@@ -162,34 +164,43 @@ def __init__(self, as_closest_canonical=False, image_only=False, dtype=np.float3
     def __call__(self, filename):
         """
         Args:
-            filename (str or file): path to file or file-like object.
+            filename (str, list, tuple, file): path file or file-like object or a list of files.
         """
-        img = nib.load(filename)
-        img = correct_nifti_header_if_necessary(img)
-
-        header = dict(img.header)
-        header['filename_or_obj'] = filename
-        header['original_affine'] = img.affine
-        header['affine'] = img.affine
-        header['as_closest_canonical'] = self.as_closest_canonical
-
-        if self.as_closest_canonical:
-            img = nib.as_closest_canonical(img)
+        if not isinstance(filename, (tuple, list)):
+            filename = (filename,)
+        img_array = list()
+        compatible_meta = dict()
+        for name in filename:
+            img = nib.load(name)
+            img = correct_nifti_header_if_necessary(img)
+            header = dict(img.header)
+            header['filename_or_obj'] = name
+            header['original_affine'] = img.affine
             header['affine'] = img.affine
+            header['as_closest_canonical'] = self.as_closest_canonical
+
+            if self.as_closest_canonical:
+                img = nib.as_closest_canonical(img)
+                header['affine'] = img.affine
+
+            img_array.append(np.array(img.get_fdata(dtype=self.dtype)))
+            img.uncache()
+
+            if len(compatible_meta) == 0:
+                for meta_key in header:
+                    meta_datum = header[meta_key]
+                    if type(meta_datum).__name__ == 'ndarray' \
```

Comment:
Why not use `isinstance` here?

Reply:
i actually copied that line from torch loader... i guess it's to avoid some incompatible subclasses of ndarray such as np.memmap